lib/api/libhoney.js

```diff
@@ -205,10 +201,30 @@ module.exports = class LibhoneyEventAPI {
         [schema.BEELINE_VERSION]: pkg.version,
         [schema.NODE_VERSION]: process.version,
       });
-      if (this.ds) {
-        ev.sampleRate = this.ds.sampleRate;
+
+      let presampled = false;
+      if (this.samplerHook) {
+        const samplerHookResponse = this.samplerHook(ev.data);
+        const keepEvent = samplerHookResponse.shouldSample;
+        if (!keepEvent) {
+          debug("skipping event due to sampler hook sample ev = " + JSON.stringify(ev.data));
+          return;
+        }
+
+        const newSampleRate = samplerHookResponse.newSampleRate;
+        if (newSampleRate !== undefined && typeof newSampleRate === "number") {
```

Comment:
can remove the `undefined` check here
```suggestion
        if (typeof newSampleRate === "number") {
```